vowpalwabbit/core/src/parser.cc

```diff
@@ -527,8 +527,9 @@ void VW::details::enable_sources(
         if (got_sigterm)
         {
           for (size_t i = 0; i < num_children; i++) { kill(children[i], SIGTERM); }
-          VW::finish(all);
-          exit(0);
+          all.finish();
+          delete &all;
```

Comment:
Is delete needed?

Reply:
Since we call `exit` immediately after we're making a best effort to free our resources. I don't think `exit` will cause destructors up the stack to call if I understand correctly.